gdal/ogr/ogrsf_frmts/ngw/ogrngwlayer.cpp

```diff
@@ -0,0 +1,1116 @@
+                    // Not supported.
+                    break;
+                case OFTString:
+                case OFTIntegerList:
+                case OFTInteger64List:
+                case OFTRealList:
+                case OFTStringList:
+                    oFieldsJson.Add( poFieldDefn->GetNameRef(),
+                        poFeature->GetFieldAsString(iField) );
+                    break;
+                case OFTDate:
+                case OFTTime:
+                case OFTDateTime:
+                {
+                    int nYear, nMonth, nDay, nHour, nMinute, nSecond, nTZFlag;
+                    if( poFeature->GetFieldAsDateTime(iField, &nYear, &nMonth,
+                        &nDay, &nHour, &nMinute, &nSecond, &nTZFlag) == TRUE)
+                    {
+                        // TODO: Convert timestamp to UTC.
+                        if( nTZFlag == 0 || nTZFlag == 100 )
+                        {
+                            CPLJSONObject oDateJson(poFieldDefn->GetNameRef(),
+                                oFieldsJson);
+
+                            oDateJson.Add("year", nYear);
+                            oDateJson.Add("month", nMonth);
+                            oDateJson.Add("day", nDay);
+                            oDateJson.Add("hour", nHour);
+                            oDateJson.Add("minute", nMinute);
+                            oDateJson.Add("second", nSecond);
+                        }
+                    }
+                    break;
+                }
+                default:
+                    break;
+            }
+        }
+    }
+
+    return oFeaturesJson;
+}
+
+/*
+ * FeatureToJsonString()
+ */
+static std::string FeatureToJsonString(OGRFeature *poFeature)
+{
+    return FeatureToJson(poFeature).Format(CPLJSONObject::Plain);
+}
+
+ /*
+  * OGRNGWLayer()
+  */
+OGRNGWLayer::OGRNGWLayer( OGRNGWDataset *poDSIn,
+    const CPLJSONObject &oResourceJsonObject ) :
+    osResourceId(oResourceJsonObject.GetString("resource/id", "-1")),
+    poDS(poDSIn),
+    bFetchedPermissions(false),
+    nFeatureCount(-1),
+    oNextPos(moFeatures.begin()),
+    nPageStart(0),
+    bNeedSyncData(false),
+    bNeedSyncStructure(false)
+{
+    std::string osName = oResourceJsonObject.GetString("resource/display_name");
+    poFeatureDefn = new OGRFeatureDefn( osName.c_str() );
+    poFeatureDefn->Reference();
+
+    poFeatureDefn->SetGeomType( NGWAPI::NGWGeomTypeToOGRGeomType(
+        oResourceJsonObject.GetString("vector_layer/geometry_type")) );
+
+    OGRSpatialReference *poSRS = new OGRSpatialReference();
+    int nEPSG = oResourceJsonObject.GetInteger("vector_layer/srs/id", 3857); // Default NGW SRS is Web mercator EPSG:3857.
+    if( poSRS->importFromEPSG( nEPSG ) == OGRERR_NONE )
+    {
+        if( poFeatureDefn->GetGeomFieldCount() != 0 )
+        {
+            poFeatureDefn->GetGeomFieldDefn(0)->SetSpatialRef( poSRS );
+        }
+    }
+    else
+    {
+        delete poSRS;
```

Comment:
remove this delete poSRS and call poSRS->Release() in all code paths (ie even when importFromEPSG() succeeds)

Reply:
Fix per 4579aeab927d70ddab6142340bf5550c6789b42e